diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -435,6 +439,12 @@ namespace diff_drive_controller{
       left_pos  /= wheel_joints_size_;
       right_pos /= wheel_joints_size_;
 
+      if (swap_front_and_back_) {
+        const double temp = left_pos;
+        left_pos = -right_pos;
+        right_pos = -temp;
+      }
+
       // Estimate linear and angular velocity using joint information
       odometry_.update(left_pos, right_pos, time);
```

Comment:
would the orientation not be affected by the swap or is it internally taken care of because of the swapping?

Reply:
Theoretically it shouldn't, but I also tested it on a robot and it was fine (you yourself actually tested the first commit, the second one is just cleanup)